mongomock/collection.py

```diff
@@ -1331,7 +1331,8 @@ def create_index(self, key_or_list, cache_for=300, session=None, **kwargs):
                 index = tuple(index)
                 if index in indexed:
                     raise DuplicateKeyError('E11000 Duplicate Key Error', 11000)
-                indexed.add(index)
+                if index:
```

Comment:
The reason this works is not obvious here. I suggest you make it explicit:
```py
if is_sparse and not index:
  continue
```
Note that you can move that above line 1331, as there is no need to convert to a tuple or check if it's in indexed either.

Reply:
yeah, your way is better